lib/jobs/get-catalog-values-job.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+var di = require('di');
+
+module.exports = getCatalogValuesJobFactory;
+di.annotate(getCatalogValuesJobFactory, new di.Provide('Job.Get.Catalog.Values'));
+di.annotate(getCatalogValuesJobFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Services.Waterline',
+    'Util',
+    '_'
+));
+
+function getCatalogValuesJobFactory(
+    BaseJob,
+    Logger,
+    waterline,
+    util,
+    _
+){
+    var logger = Logger.initialize(getCatalogValuesJobFactory);
+
+    /**
+     * This job finds one or more requested values from one or more specified catalogs.
+     * The catalog values are made available in the shared context in an object that 
+     * maps the catalog values to requester provided keys.
+     *
+     * Example requestedData array:
+     * [{
+     *   "source": "ohai",
+     *   "keys": {
+     *     "myIPAddrKeyName": "data.ipaddress"
+     *   }
+     * }]
+     *
+     * Adds to context.data:
+     * {
+     *   "myIPAddrKeyName": <catalog value>
+     * }
+     *
+     * @param options
+     * @param context
+     * @param taskId
+     * @constructor
+     */
+
+    function GetCatalogValuesJob(options, context, taskId){
+        GetCatalogValuesJob.super_.call(this, logger, options, context, taskId);
+        this.nodeId = context.target;
+        this.options = options;
+        this.logger = logger;
+    }
+
+    util.inherits(GetCatalogValuesJob, BaseJob);
+
+    /**
+     * @memberOf GetCatalogValuesJob
+     */
+    GetCatalogValuesJob.prototype._run = function _run(){
+        var self = this,
+            newObj = {};
+
+        // Get the correct catalog based on source
+        _.forEach(self.options.requestedData, function(catalogInfo){
```

Comment:
@yyscamper I was hoping you could expound because I'm not actually seeing the same behavior that you are indicating. When I provide multiple objects in the requestedData array, the `_.forEach` successfully iterates over them and returns all the requested data.

Could you also clarify why you are recommending using `Promise.reduce` rather than the map? I might be misunderstanding the documentation for reduce but it doesn't seem to line up with what the job is trying to accomplish. The job will map the located catalog values to the user provided keys.

Given the input as the requstedData array:
```
[
    {
		"source": "ohai",
		"keys"{
			"myDefinedKey": "data.ipaddress",
			"myOtherDefinedKey": "data.cpu.0.cores"
		}
	},
	{
		"source": "lshw",
		"keys": {
			"randomName": "data.handle"
	}
]
```
It would return: 
```
{
	"myDefinedKey": "1.1.1.1",
	"myOtherDefinedKey": "1",
	"randomName": "DMI:0001"
}
```

I am open to suggestions on how to simplify this section because I know it is a bit complex.